typeutils/unsafe.go

```diff
@@ -11,20 +11,24 @@ import (
 // passed as argument are changed while the string is used.  No checking whether
 // bytes are valid UTF-8 data is performed.
 func BytesToString(b []byte) string {
+	// ensure the underlying bytes don't get GC'ed before the assignment happens
+	defer runtime.KeepAlive(&b)
```

Comment:
Is there a particular reason for `runtime.KeepAlive`?
`*(*string)(unsafe.Pointer(&b))` should be "atomic"